packages/codelabs/src/basics/lit-html.md

```diff
@@ -0,0 +1,1139 @@
+    </script>
+  </body>
+</html>
+```
+
+Make sure you add `type="module"` to the script tag.
+
+Next, we need to define our web component. When writing a vanilla web component we would extend from `HTMLElement`. With lit-element we need to import and extend from `LitElement`, which itself is an extension of `HTMLElement`.
+
+```js
+class TodoApp extends LitElement {
+  connectedCallback() {
+    super.connectedCallback();
+    console.log('lit element connected');
+  }
+}
+
+customElements.define('todo-app', TodoApp);
+```
+
+We defined the tag name for our element as `todo-app`, now we need to add this to the HTML of our page:
+
+```html
+<todo-app></todo-app>
+```
+
+If you run this in the browser you should see `lit element connected` logged to the terminal.
+
+<aside class="notice">
+  Because LitElement also does some work in the connectedCallback, we must always call super.connectedCallback. This is a common source of confusion, so make sure to remember!
+</aside>
+
+Now that we have defined our element, we can start adding a template. lit-element uses lit-html to handle the templating. lit-html works by writing HTML inside of [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) which are strings that can span multiple lines:
+
+```js
+const template = `
+ <h1>Hello world</h1>
+`;
+```
+
+In order to create an actual `lit-html` template, we prefix the template literal with a special HTML tag:
+
+```js
+import { html } from 'https://unpkg.com/lit-element?module';
+
+const template = html`
+  <h1>Hello world</h1>
+`;
+```
+
+This is using a feature called [tagged template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Tagged_templates), where the `html` tag is a function which returns the prepared template ready for rendering. We won't go into details of how it works exactly, but by using this syntax `lit-html` can very efficiently update the dynamic parts of your template when your element re-renders.
+
+Most popular IDEs support syntax highlighting of HTML inside template literals, for some you might need to install a plugin. [See our IDE section](https://open-wc.org/developing/ide.html#visual-studio-code) to learn more about that.
+
+lit-element works with a `render` function, which is called each time the element is updated. From this function, we return the template which is rendered to the page. Let's start by displaying the title of our app:
+
+```js
+class TodoApp extends LitElement {
+  render() {
+    return html`
+      <h1>Todo app</h1>
+    `;
+  }
+}
+
+customElements.define('todo-app', TodoApp);
+```
+
+If you refresh the browser, you should see the title displayed on the page.
+
+<detail>
+  <summary>View final result</summary>
+
+```html
+<!DOCTYPE html>
+<html>
+  <head></head>
+
+  <body>
+    <todo-app></todo-app>
+
+    <script type="module">
+      ...code sample goes here...
```

Comment:
is this a to-do?

Reply:
yep need to gather the final code results of all the steps